app/models/role_loader.rb

```diff
@@ -0,0 +1,45 @@
+class RoleLoader
+
+  def available_roles_for_application(name)
```

Comment:
I'm thinking if we should ever think about roles per application. Shouldn't it only ever be by application type? Or application + organisation type.